### ISSUE TITLE
Sound and music playback

### ISSUE THREAD

**Issue by klangtuch** [deadcast2/UltraEd#153]
UltraEd is currently to my knowledge not able to play any kind of sound. From my limited research the N64 seems to use ADPCM for samples.
https://ultra64.ca/files/documentation/online-manuals/man/pro-man/pro19/index.html

I'm not sure if the music playback needs a completly different implementation vs the sound effects.
If so, it's possibly better to split this issue into sound playback and music playback.
So in short:

1. Find out how sound in the N64 works
2. If different implementations are needed, split this issue into sound and music playback
3. Implementation

That would be a major milestone, since it makes it possible to create a nice demo game to show off the capabilities of this engine.

**Comment by deadcast2** [comment]
Thanks for looking into this. Yeah there's currently no audio support in UltraEd. You're right about the audio format used. I made a proof-of-concept demo years ago testing the audio playback capability of the system. I only tested playing audio samples for quick sound effects and one longer music track. Didn't try out the midi system which is the preferred way for long music tracks. I don't remember it being too hard to actually setup since libultra takes care of a lot of the work. I recall the official SDK including the compressor that audio files must first be run through. UltraEd would have to "process" imported audio files using the SDK's compressor. 

I agree having sound would really start to make UltraEd a more serious tool. I'm pretty occupied right now with a small project that I'm trying to finish up. Juggling too many things lol. Fell down the rabbit hole of collision detection and response. The info I'm learning may be helpful for UltraEd. So not sure when I'll be able to work on this just yet.
